helix-term/src/commands/engine/scheme.rs

```diff
@@ -0,0 +1,2003 @@
+struct LanguageConfigurationContainer {
+    configuration: Rc<RefCell<Option<toml::Value>>>,
+}
+
+impl LanguageConfigurationContainer {
+    fn new() -> Self {
+        Self {
+            configuration: Rc::new(RefCell::new(None)),
+        }
+    }
+}
+
+impl Custom for LanguageConfigurationContainer {}
+
+impl LanguageConfigurationContainer {
+    fn add_configuration(&self, config_as_string: String) -> Result<(), String> {
+        let left = self
+            .configuration
+            .replace(Some(toml::Value::Boolean(false)));
+
+        if let Some(left) = left {
+            let right = serde_json::from_str(&config_as_string).map_err(|err| err.to_string());
+
+            // panic!("{:#?}", right);
+
+            match right {
+                Ok(right) => {
+                    self.configuration
+                        .replace(Some(merge_toml_values(left, right, 3)));
+
+                    Ok(())
+                }
+                Err(e) => Err(e),
+            }
+        } else {
+            let right = serde_json::from_str(&config_as_string).map_err(|err| err.to_string())?;
+
+            self.configuration.replace(Some(right));
+
+            Ok(())
+        }
+    }
+
+    fn as_language_configuration(&self) -> Option<Result<Configuration, toml::de::Error>> {
+        if let Some(right) = self.configuration.borrow().clone() {
+            let config = helix_loader::config::user_lang_config();
+
+            let res = config
+                .map(|left| merge_toml_values(left, right, 3))
+                .and_then(|x| x.try_into());
+
+            Some(res)
+            // )
+        } else {
+            None
+        }
+    }
+
+    fn get_language_configuration() -> Option<Result<Configuration, toml::de::Error>> {
+        LANGUAGE_CONFIGURATIONS.with(|x| x.as_language_configuration())
+    }
+}
+
+struct ThemeContainer {
+    themes: Rc<RefCell<HashMap<String, Theme>>>,
+}
+
+impl Custom for ThemeContainer {}
+
+impl ThemeContainer {
+    fn new() -> Self {
+        Self {
+            themes: Rc::new(RefCell::new(HashMap::new())),
+        }
+    }
+
+    fn get(name: &str) -> Option<Theme> {
+        THEME_MAP.with(|x| x.themes.borrow().get(name).cloned())
+    }
+
+    fn names() -> Vec<String> {
+        THEME_MAP.with(|x| x.themes.borrow().keys().cloned().collect())
+    }
+
+    fn register(name: String, theme: Theme) {
+        THEME_MAP.with(|x| x.themes.borrow_mut().insert(name, theme));
+    }
+}
+
+fn load_language_configuration_api(engine: &mut Engine) {
+    let mut module = BuiltInModule::new("helix/core/languages".to_string());
+
+    module.register_fn(
+        "register-language-configuration!",
+        |language_configuration: String| -> Result<(), String> {
+            LANGUAGE_CONFIGURATIONS.with(|x| x.add_configuration(language_configuration))
+        },
+    );
+
+    module.register_fn("flush-configuration", refresh_language_configuration);
```

Comment:
This was just an artifact of the old way I was storing the configuration. This is no longer needed since now steel just manipulates a reference to the configuration object rather than applying a built up queue of changes.